Fix zk assets

ArnSch opened the pull request:
**Note: This PR should not be merged unless automatic release to NPM is disabled.**

## Description

This pull request addresses two points:

1. Our zkAssetOwnable used to implement its own ownership logic, and now inherits from OpenZepplin's Ownable.sol
2. Our previous zkAsset contracts made it difficult for a user looking at deployed source code to tell whether the asset they were interacting with could adjust it's total supply. This PR makes this simpler by hardcoding the _canAdjustSupply flags based on the type of asset deployed, and by making the _canConvert be derived from the _linkedTokenAddress.

## Testing instructions

`npm run test`

I've added two test cases to zkAsset, both testing that the implicit setting of the canConvert flag given the value of _linkedTokenAddress behaves appropriately: `should set canConvert flag to false if 0x0 is linked token address` and `should set canConvert flag to true if linked token address is provided`

## Types of changes

* Bug fix (non-breaking change which fixes an issue)

* Breaking change (fix or feature that would cause existing functionality to change)

Commit:
fix(protocol): make zkAssetOwnable inherit from OpenZepplin Ownable

Commit:
fix(protocol): make zkAsset templates enforce flags appropriate to type and derive _canConvert implicitly from linked token address

Commit:
style(protocol): Move zkAsset templates to relevant folders

Commit:
add two test cases for implicit canConvert flag setting

Commit:
fix(protocol): remove unused variables

Commit:
fix(protocol): rm package-locks

Commit:
fix(protocol): re-add .env.example

Commit:
style(protocol): update docstring

zac-williamson reviewed (approved):
👍

Commit:
style(protocol): move visibility modifier to be first on construcotr

thomas-waite commented on packages/protocol/migrations/4_deploy_zk_asset.js:
```diff
@@ -26,9 +26,9 @@ module.exports = (deployer, network) => {
-                canAdjustSupply,
-                canConvert
+                ERC20_SCALING_FACTOR
+                // canAdjustSupply,
```
Think we can just remove // canAdjustSupply and // canConvert

thomas-waite committed:
refactor(protocol): fix rebase problems
update ZkAssetBase.sol and ZkAssetOwnableBase.sol  with recent improvements and changes to ZkAsset.sol and ZkAssetOwnableBase.sol
move contracts to current directory structure

thomas-waite committed:
test(protocol): fix rebase problems
update tests in ERC1724 folder with tests modified on fix-ZkAssets branch in ZkAsset folder
remove tests in out of date directory, ZkAsset
